Release/1.0.22

phillwiggins opened the pull request:
@rodrigosmarques 
@yulingtianxia 

I have merged master (as people have been committing work there), to the latest release branch. Please can you (if your free and available) test the latest version and ensure everything is working.

From my project, nearly everything is looking okay. Couple of small bugs but something that shouldn't affect production code for any other users.

rodrigosmarques commented:
Hi @phillwiggins 
My project does not run. Problem persists on sembast.
I deleted the github cache, deleted the build folder, deleted the app on the device.
```dart
  parse_server_sdk:
    git: 
      url: https://github.com/phillwiggins/flutter_parse_sdk.git
      ref: release/1.0.22
--------------------------------------------------------------------

Launching lib/main.dart on iPhone Xʀ in debug mode...
Xcode build done.                                           16,6s
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: NoSuchMethodError: The getter 'length' was called on null.
Receiver: null
Tried calling: length
#0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:50:5)
#1      XXTEA._fixkey
#2      XXTEA.encrypt
#3      XXTEA.encryptToString
#4      _XXTeaEncoder.convert
#5      Codec.encode  (dart:convert/codec.dart:21:32)
#6      getCodecEncodedSignature
#7      SembastDatabase.open.<anonymous closure>._openDone
<asynchronous suspension>
#8      SembastDatabase.open.<anonymous closure>
<asynchronous suspension>
#9      BasicLock.synchronized
<asynchronous suspension>
#10     SembastDatabase.open (p<…>

```

WahdanZ commented:
I have some comments, now we will make sembast the default storage I think it will be okay
but when I implemented it the first time I used the master key as the encryption key and I modified this in the next PR and make the appID to be the encryption key when we make sharedprefrances as the default implementation  
you can see the PR here https://github.com/phillwiggins/flutter_parse_sdk/pull/168

rodrigosmarques commented:
Hi @WahdanZ 
changing the sembast to default, what will happen to the applications that store the information in the SharedPreference?
What is the plan for the developer to switch from SharedPreference to Sembast. There are applications in production today. How will the data be taken from one storage to another?

WahdanZ commented:
yes I understand that and that's why I make SharedPreference it the default one

rodrigosmarques commented:
Ok @WahdanZ.
I have not had time to examine the code, but it does not seem that SharedPreference is by default. The error message is related to Sebast.

WahdanZ commented:
yes because there are some comments ignored  
https://github.com/phillwiggins/flutter_parse_sdk/pull/168

phillwiggins commented:
Why don't we introduce a book to define shared preferences if necessary but
use Sembast as default

On Mon, Jun 10, 2019, 23:53 Ahmed Wahdan <notifications@github.com> wrote:

> yes because there are some comments ignored
> #168 <https://github.com/phillwiggins/flutter_parse_sdk/pull/168>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/phillwiggins/flutter_parse_sdk/pull/193?email_source=notifications&email_token=AB4CPXRQIEHBKA3VC4CSDZDPZ3LOBA5CNFSM4HWRJAH2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXLONYY#issuecomment-500623075>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB4CPXWZNCSFVISF7UB7EJLPZ3LOBANCNFSM4HWRJAHQ>
> .
>

rodrigosmarques commented:
Ok @phillwiggins 
This can be used. My point is that SharedPreference exists from the beginning. There are applications in production using.
Using sembast by default, we can break apps and an inattentive developer does not realize this change.
The lib documentation has to be improved at this point.
Before making this kind of change, I suggest always mark it as @Deprecated.

rodrigosmarques commented:
@phillwiggins 

Exactly. I understood that this was what they had defined e done here #168 by @WahdanZ 

#[](https://github.com/phillwiggins/flutter_parse_sdk/pull/167#issuecomment-488673448)

phillwiggins commented:
Okay, so this current version should be working then.

Are tests running okay on your project?

On Tue, 11 Jun 2019 at 16:38, Rodrigo de Souza Marques <
notifications@github.com> wrote:

> @phillwiggins <https://github.com/phillwiggins>
>
> Exactly. I understood that this was what they had defined e done here #168
> <https://github.com/phillwiggins/flutter_parse_sdk/pull/168> by @WahdanZ
> <https://github.com/WahdanZ>
>
> #
> <https://github.com/phillwiggins/flutter_parse_sdk/pull/167#issuecomment-488673448>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/phillwiggins/flutter_parse_sdk/pull/193?email_source=notifications&email_token=AB4CPXVSEGQIXS5JCZL7YD3PZ7BHZA5CNFSM4HWRJAH2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXNRXII#issuecomment-500898721>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB4CPXRPHJFVWEVR3T3N3YDPZ7BHZANCNFSM4HWRJAHQ>
> .
>


-- 
Kind Regards

Phill Wiggins
Phill.Wiggins@Gmail.com

rodrigosmarques commented:
@phillwiggins 
Have you done any merge? It was not working for me.
I deleted the github cache, deleted the build folder, deleted the app on the device.
'''dart
 parse_server_sdk:
    git: 
      url: https://github.com/phillwiggins/flutter_parse_sdk.git
      ref: release/1.0.22
--------------------------------------------------------------------

Launching lib/main.dart on iPhone Xʀ in debug mode...
Xcode build done.                                           16,6s
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: NoSuchMethodError: The getter 'length' was called on null.
Receiver: null
Tried calling: length
#0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:50:5)
#1      XXTEA._fixkey
#2      XXTEA.encrypt
#3      XXTEA.encryptToString
#4      _XXTeaEncoder.convert
#5      Codec.encode  (dart:convert/codec.dart:21:32)
#6      getCodecEncodedSignature
#7      SembastDatabase.open.<anonymous closure>._openDone
<asynchronous suspension>
#8      SembastDatabase.open.<anonymous closure>
<asynchronous suspension>
#9      BasicLock.synchronized
<asynchronous suspension>
#10     SembastDatabase.open (p<…>
'''

phillwiggins commented:
Try again now?

This is not a proper fix though. This just moves the default from Sembast
to SharedPrefs so there is still an issue here.

On Tue, 11 Jun 2019 at 17:53, Rodrigo de Souza Marques <
notifications@github.com> wrote:

> @phillwiggins <https://github.com/phillwiggins>
> Have you done any merge? It was not working for me.
> I deleted the github cache, deleted the build folder, deleted the app on
> the device.
> '''dart
> parse_server_sdk:
> git:
> url: https://github.com/phillwiggins/flutter_parse_sdk.git
> ref: release/1.0.22
>
> Launching lib/main.dart on iPhone Xʀ in debug mode...
> Xcode build done. 16,6s
> [VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: NoSuchMethodError:
> The getter 'length' was called on null.
> Receiver: null
> Tried calling: length
> #0 Object.noSuchMethod (dart:core-patch/object_patch.dart:50:5)
> #1 <https://github.com/phillwiggins/flutter_parse_sdk/pull/1>
> XXTEA._fixkey
> #2 <https://github.com/phillwiggins/flutter_parse_sdk/issues/2>
> XXTEA.encrypt
> #3 <https://github.com/phillwiggins/flutter_parse_sdk/issues/3>
> XXTEA.encryptToString
> #4 <https://github.com/phillwiggins/flutter_parse_sdk/pull/4>
> _XXTeaEncoder.convert
> #5 <https://github.com/phillwiggins/flutter_parse_sdk/pull/5>
> Codec.encode (dart:convert/codec.dart:21:32)
> #6 <https://github.com/phillwiggins/flutter_parse_sdk/pull/6>
> getCodecEncodedSignature
> #7 <https://github.com/phillwiggins/flutter_parse_sdk/pull/7>
> SembastDatabase.open.._openDone
>
> #8 <https://github.com/phillwiggins/flutter_parse_sdk/issues/8>
> SembastDatabase.open.
>
> #9 <https://github.com/phillwiggins/flutter_parse_sdk/pull/9>
> BasicLock.synchronized
>
> #10 <https://github.com/phillwiggins/flutter_parse_sdk/issues/10>
> SembastDatabase.open (p<…>
> '''
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/phillwiggins/flutter_parse_sdk/pull/193?email_source=notifications&email_token=AB4CPXRDQSSNOM4XOEYW2XLPZ7KBPA5CNFSM4HWRJAH2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXNZRFI#issuecomment-500930709>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB4CPXU6M64W24UDFEI2WD3PZ7KBPANCNFSM4HWRJAHQ>
> .
>


-- 
Kind Regards

Phill Wiggins
Phill.Wiggins@Gmail.com

rodrigosmarques commented:
Hi @phillwiggins 
Hello phill.
I performed the test and the same error continues.
To make sure that using the correct version, I cleaned the cache of the git's repository in the flutter-cache.
I deleted the app on the device, and I ran flutter clean
```dart
  parse_server_sdk:
    git: 
      url: https://github.com/phillwiggins/flutter_parse_sdk.git
      ref: release/1.0.22
-----------
Launching lib/main.dart on iPhone Xʀ in debug mode...
Xcode build done.                                           27,1s
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: type 'Future<CoreStore>' is not a subtype of type 'CoreStore'
#0      ParseCoreData.init
#1      Parse.initialize
#2      main
#3      _AsyncAwaitCompleter.start  (dart:async-patch/async_patch.dart:49:6)
#4      main
#5      _runMainZoned.<anonymous closure>.<anonymous closure>  (dart:ui/hooks.dart:199:25)
#6      _rootRun  (dart:async/zone.dart:1124:13)
#7      _CustomZone.run  (dart:async/zone.dart:1021:19)
#8      _runZoned  (dart:async/zone.dart:1516:10)
#9      runZoned  (dart:async/zone.dart:1500:12)
#10     _runMainZoned.<anonymous closure>  (dart:ui/hooks.dart:190:5)
#11     _startIsolate.<anonymous closure>  (dart:isolate-patch/isolate_patch.dart:300:19)
#12     _RawReceivePortImpl._handleMessage  (dart:isolate-patch/isolate_patch.dart:171:12)
```

phillwiggins commented:
Okay cool. Thank you.

I will try to fix, however, I don't own an iPhone or Mac so will struggle
to test.

On Thu, Jun 13, 2019, 04:20 Rodrigo de Souza Marques <
notifications@github.com> wrote:

> Hi @phillwiggins <https://github.com/phillwiggins>
> Hello phill.
> I performed the test and the same error continues.
> To make sure that using the correct version, I cleaned the cache of the
> git's repository in the flutter-cache.
> I deleted the app on the device, and I ran flutter clean
>
>   parse_server_sdk:
>
>     git:
>
>       url: https://github.com/phillwiggins/flutter_parse_sdk.git
>
>       ref: release/1.0.22
> -----------
> Launching lib/main.dart on iPhone Xʀ in debug mode...
> Xcode build done.                                           27,1s
>
> [VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: type 'Future<CoreStore>' is not a subtype of type 'CoreStore'
>
> #0      ParseCoreData.init
>
> #1      Parse.initialize
>
> #2      main
>
> #3      _AsyncAwaitCompleter.start  (dart:async-patch/async_patch.dart:49:6)
>
> #4      main
>
> #5      _runMainZoned.<anonymous closure>.<anonymous closure>  (dart:ui/hooks.dart:199:25)
>
> #6      _rootRun  (dart:async/zone.dart:1124:13)
>
> #7      _CustomZone.run  (dart:async/zone.dart:1021:19)
>
> #8      _runZoned  (dart:async/zone.dart:1516:10)
>
> #9      runZoned  (dart:async/zone.dart:1500:12)
>
> #10     _runMainZoned.<anonymous closure>  (dart:ui/hooks.dart:190:5)
>
> #11     _startIsolate.<anonymous closure>  (dart:isolate-patch/isolate_patch.dart:300:19)
>
> #12     _RawReceivePortImpl._handleMessage  (dart:isolate-patch/isolate_patch.dart:171:12)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/phillwiggins/flutter_parse_sdk/pull/193?email_source=notifications&email_token=AB4CPXXDFY3KWVK6QJKUHCDP2G4GHA5CNFSM4HWRJAH2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXSMTQY#issuecomment-501533123>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB4CPXVX2O5E35MPDVW5CM3P2G4GHANCNFSM4HWRJAHQ>
> .
>

phillwiggins commented:
This looks like an obvious fix though. It's simply looking like a Future is
being expected but we are creating an instance of CoreStore.

On Thu, Jun 13, 2019, 04:20 Rodrigo de Souza Marques <
notifications@github.com> wrote:

> Hi @phillwiggins <https://github.com/phillwiggins>
> Hello phill.
> I performed the test and the same error continues.
> To make sure that using the correct version, I cleaned the cache of the
> git's repository in the flutter-cache.
> I deleted the app on the device, and I ran flutter clean
>
>   parse_server_sdk:
>
>     git:
>
>       url: https://github.com/phillwiggins/flutter_parse_sdk.git
>
>       ref: release/1.0.22
> -----------
> Launching lib/main.dart on iPhone Xʀ in debug mode...
> Xcode build done.                                           27,1s
>
> [VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: type 'Future<CoreStore>' is not a subtype of type 'CoreStore'
>
> #0      ParseCoreData.init
>
> #1      Parse.initialize
>
> #2      main
>
> #3      _AsyncAwaitCompleter.start  (dart:async-patch/async_patch.dart:49:6)
>
> #4      main
>
> #5      _runMainZoned.<anonymous closure>.<anonymous closure>  (dart:ui/hooks.dart:199:25)
>
> #6      _rootRun  (dart:async/zone.dart:1124:13)
>
> #7      _CustomZone.run  (dart:async/zone.dart:1021:19)
>
> #8      _runZoned  (dart:async/zone.dart:1516:10)
>
> #9      runZoned  (dart:async/zone.dart:1500:12)
>
> #10     _runMainZoned.<anonymous closure>  (dart:ui/hooks.dart:190:5)
>
> #11     _startIsolate.<anonymous closure>  (dart:isolate-patch/isolate_patch.dart:300:19)
>
> #12     _RawReceivePortImpl._handleMessage  (dart:isolate-patch/isolate_patch.dart:171:12)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/phillwiggins/flutter_parse_sdk/pull/193?email_source=notifications&email_token=AB4CPXXDFY3KWVK6QJKUHCDP2G4GHA5CNFSM4HWRJAH2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXSMTQY#issuecomment-501533123>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB4CPXVX2O5E35MPDVW5CM3P2G4GHANCNFSM4HWRJAHQ>
> .
>

WahdanZ commented:
I faced this problem and I'm going to make PR for it today 
also I add an option to Create Parse object without data it will be useful for pointer it's and feature in Parse Android sdk
